packages/ckb/src/utils/rgbpp.ts

```diff
@@ -219,3 +226,28 @@ export const throwErrorWhenTxInputsExceeded = (inputLen: number) => {
     throw new RgbppCkbTxInputsExceededError(`Please ensure the tx inputs do not exceed ${RGBPP_TX_INPUTS_MAX_LENGTH}`);
   }
 };
+
+// Check the validity of RGB++ cells and throw an exception if the conditions are not met to avoid building invalid CKB TX
+export const throwErrorWhenRgbppCellsInvalid = (
+  cells: IndexerCell[] | undefined,
+  xudtTypeBytes: Hex,
+  isMainnet: boolean,
+) => {
+  if (!cells || cells.length === 0) {
+    throw new NoRgbppLiveCellError('No rgbpp cells found with the rgbpp lock args');
+  }
+  const typeCells = cells.filter((cell) => !!cell.output.type);
+  if (typeCells.length === 0) {
+    throw new NoRgbppLiveCellError('No rgbpp cells found with the rgbpp lock args');
+  }
+  const isSporeExist = typeCells.some((cell) => isScriptPartialEqual(cell.output.type!, getSporeTypeScript(isMainnet)));
+  if (isSporeExist) {
+    throw new RgbppUtxoBindMultiTypeAssetsError(
+      'The BTC UTXO must not be bound to Spore and xUDT cells at the same time',
+    );
+  }
```

Comment:
Some nonUDTType and nonClusterSpore may exist.
I think using `isTypeAssetSupported` to check all the cells is more appropriate.

Reply:
I used the function `isUDTTypeSupported` to filter non-xUDT assets for RGB++ xUDT transfer and leap in https://github.com/ckb-cell/rgbpp-sdk/pull/139/commits/b009bd7baa34199cc3115588d8d107b3035c61a3